pkg/docker/docker.go

```diff
@@ -609,15 +610,30 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	containerName := containerNameOrID(container)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+	removalDone := make(chan struct{}, 1)
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := d.RemoveContainer(container.ID); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
+		removalDone <- struct{}{}
+	}
+	go func() {
+		sigchan := make(chan os.Signal, 1)
+		signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
```

Comment:
this isn't in origin, are you suggesting i vendor your interrupt package from k8s into s2i?  feels a little crazy.